### PR TITLE
Add Watchdog verbose logging

### PR DESCRIPTION
**Pull request by MAGICCC**
Adding the possibility to see verbose logs while sending mails via Watchdog
May fix https://github.com/mailcow/mailcow-dockerized/issues/4298 + https://github.com/mailcow/mailcow-dockerized/issues/4197

PS: Needs a new watchdog image then
PPS: Needs more testing - only tested it quickly. @andryyy maybe you have some time for this

### CODE REVIEW

File: data/Dockerfiles/watchdog/watchdog.sh [hunk 1]
```diff
@@ -133,7 +140,11 @@ function mail_error() {
       --hello-host=${MAILCOW_HOSTNAME} \
       --ipv4
       #--server="${RCPT_MX}"
-    log_msg "Sent notification email to ${rcpt}"
+    if [[ $? -eq 1 ]]; then # exit code 1 is fine
```
Comment (andryyy) [inline data/Dockerfiles/watchdog/watchdog.sh:146]: Looks good in my test, but why? Did you find an issue regarding this? Is it normal?

**Comment by andryyy** [comment]
Nice. :) Let me know when you can merge. I will build the image then first.

File: data/Dockerfiles/watchdog/watchdog.sh (hunk 1 above)
Comment (MAGICCC) [reply data/Dockerfiles/watchdog/watchdog.sh:146]: Looks like it. I found https://github.com/mludvig/smtp-cli/pull/17 and seems exitcode is 1 when everything is fine...

**Comment by MAGICCC** [comment]
I just tested a bit and looks fine on my side, but I noticed when you set a mail with a bad domain (no mx record), then no mail will be sent.
Maybe we can reuse this: https://github.com/mailcow/mailcow-dockerized/blob/8b4e1ce70f4724243bf748a4ad231e18ef2732f2/data/Dockerfiles/watchdog/watchdog.sh#L129-L135

like this:
```
RCPT_DOMAIN=$(echo ${rcpt} | awk -F @ {'print $NF'})
CHECK_FOR_VALID_MX=$(dig +short ${RCPT_DOMAIN} mx | sort -n | awk '{print $2; exit}')
if [[ -z ${CHECK_FOR_VALID_MX} ]]; then
  log_msg "Cannot determine MX for ${rcpt}, skipping email notification..."
  return 1
fi
```

**Comment by andryyy** [comment]
`CHECK_FOR_VALID_MX=$(dig +short ${RCPT_DOMAIN} mx)` would be enough. :)

Can you adjust the PR?

**Comment by andryyy** [comment]
Thank youuu

**Comment by andryyy** [comment]
We should probably add the variable information to the docs.

**Comment by dragoangel** [comment]
Wow, nice :)

**Pull request merged**